wordstats/cognate_info.py

```diff
@@ -0,0 +1,186 @@
+        new_registry.candidates = load_from_path(language_code_path).splitlines()
+
+        return new_registry
+
+    def save_candidates(self):
+
+        language_code_path = path_of_cognate_candidates(self.language_ids, self.method_name)
+        save_to_file(language_code_path, '\n'.join(self.candidates))
+
+    def save_evaluation(self):
+
+        language_code_path = path_of_cognate_whitelist(self.language_ids)
+        save_to_file(language_code_path, '\n'.join(self.whitelist))
+
+        language_code_path = path_of_cognate_blacklist(self.language_ids)
+        save_to_file(language_code_path, '\n'.join(self.blacklist))
+
+    # generates candidates based on distance function func and word lists
+    def apply_distance_metric(self, wordlist1, wordlist2, func):
+
+        for w1 in wordlist1:
+            newlist = list(filter(lambda x: len(x)>0 , map(lambda w: w1 + ' ' + w if func(w1, w) else "", wordlist2)))
+            self.candidates.extend(newlist)
+
+    # note: only executable in terminal, reactive to one-key stroke
+    def start_quiz(self):
+        print(self.language_ids)
+        print("y:   whitelist")
+        print("q:   quit")
+        print("other:   blacklist")
+
+        for cognate in self.candidates:
+            print(cognate)
+
+            char = read_single_keypress()
+
+            if char == 'y':
+                self.whitelist.append(cognate)
+            elif char == 'q':
+                break
+            else:
+                self.blacklist.append(cognate)
+
+
+
+
+    # TODO: implement load and cache from /to db
+    @classmethod
+    def load_from_db(cls, cognate_code, method_name= ""):
+        """
+        Assumes the ~./word_info file contains
+        information about how to connect to the
+        database. Also, assumes the database contains
+        the information.
+
+        :param language_id:
+        :return:
+        """
+
+        new_registry = cls(cognate_code, method_name= "")
+
+        all_word_info_items = WordInfo.find_all(cognate_code)
+
+        for each in all_word_info_items:
+            new_registry.word_info_dict[each.word] = each
+
+        return new_registry
+
+    def cache_to_db(self):
+        """
+        Useful to save the informations about a given language
+        in the DB. Loading from the DB is faster than from file.
+        See the tests file
+        :return:
+        """
+
+        def clear_corresponding_entries_in_db(self):
+            table = Table('word_info', Base.metadata, autoload=True, autoload_with=BaseService.engine)
+            words = BaseService.session.query(table).filter(table.c.language_id == self.language_id)
+            words.delete(synchronize_session=False)
+
+        clear_corresponding_entries_in_db(self)
+        for word_info in list(self.word_info_dict.values()):
+            BaseService.session.add(word_info)
+        BaseService.session.commit()
+
+    # Everything that follows is private
```

Comment:
if it's private, the convention is to have it named with an underscore as prefix. e.g. `_print_stats`